src/main/java/org/fenixedu/academic/domain/student/curriculum/calculator/CreditsWeightedCalculator.java

```diff
@@ -0,0 +1,50 @@
+package org.fenixedu.academic.domain.student.curriculum.calculator;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+
+import org.fenixedu.academic.domain.exceptions.DomainException;
+import org.fenixedu.academic.util.Bundle;
+import org.fenixedu.bennu.core.i18n.BundleUtil;
+
+public class CreditsWeightedCalculator extends CreditsWeightedCalculator_Base {
+
+    private CreditsWeightedCalculator() {
+        super();
+    }
+
+    public static CreditsWeightedCalculator createCreditsWeightedCalculator() {
```

Comment:
Is this method used? We should avoid (if possible) multiple creator methods for the same entity.